chain/store/stateGenerator.go

```diff
@@ -147,6 +147,29 @@ func (cs *ChainStore) spendTransaction(states *StateDB, txn *transaction.Transac
 			return err
 		}
 
+		if err = states.UpdateBalance(donationAddress, config.NKNAssetID, common.Fixed64(genID.RegistrationFee), Addition); err != nil {
+			return err
+		}
+	case pb.PayloadType_GENERATE_ID_2_TYPE:
+		genID := pl.(*pb.GenerateID2)
+
+		if err = states.UpdateBalance(common.BytesToUint160(genID.Sender), config.NKNAssetID, common.Fixed64(genID.RegistrationFee), Subtraction); err != nil {
+			return err
+		}
+
+		IDpg, err := program.CreateProgramHash(genID.PublicKey)
```

Comment:
local variable should start with non-capital letter, use `idPg` or just `pg` instead